File: frontend/js/media.js
import { get, post } from "./api.js";

import { emit } from "./events.js";


export async function loadMedia() {
    const data = await get("/media");

    if (data && !data.error)
        emit("media:update", data);

    return data;
}


export async function mediaAction(action, position) {
    const payload = { action: action };

    if (position !== undefined)
        payload.position = position;

    return post("/media/action", payload);
}

File: routes/media.py
from flask import Blueprint
from ._errors import envelope

from services.media import (
    get_media,
    media_action
)


media_bp = Blueprint(
    "media",
    __name__
)



@media_bp.route(
"",
methods=["GET"]
)
def current_media():
    return envelope(get_media(), default_code="ERR_MEDIA_STATUS")



@media_bp.route(
"/action",
methods=["POST"]
)
def action():

    from flask import request

    data = request.json or {}
    return envelope(media_action(data.get("action"), data.get("position")), default_code="ERR_MEDIA_ACTION")